### PR TITLE
HIVE-26636 - Adding hyperbolic functions

### PR DESCRIPTION
**Pull request by ggangadharan**
### What changes were proposed in this pull request?

Added Hyperbolic functions 

cosh
sinh
tanh

### Why are the changes needed?

Added hyperbolic built-in functions for MATH usage. 

### Does this PR introduce _any_ user-facing change?

No  

### How was this patch tested?

Added unit test case in the TestUDFMath file. Also tested the results through udf_*.q files

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java [hunk 1]
```diff
@@ -32,6 +32,7 @@
 import java.util.regex.Pattern;
 import java.util.regex.PatternSyntaxException;
 
+import org.apache.hadoop.hive.ql.udf.*;
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java:35]: please avoid wildcard imports

File: ql/src/java/org/apache/hadoop/hive/ql/udf/UDFCosh.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedExpressions;
+import org.apache.hadoop.hive.ql.exec.vector.expressions.gen.FuncCoshDoubleToDouble;
+import org.apache.hadoop.hive.ql.exec.vector.expressions.gen.FuncCoshLongToDouble;
+import org.apache.hadoop.hive.serde2.io.DoubleWritable;
+
+/**
+ * UDFCosh.
+ */
+@Description(name = "cosh",
+        value = "_FUNC_(x) - returns the cosineh of x (x is in radians)",
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/udf/UDFCosh.java:30]: returns the `hyperbolic cosine` ?

File: ql/src/java/org/apache/hadoop/hive/ql/udf/UDFTanh.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+package org.apache.hadoop.hive.ql.udf;
+
+import org.apache.hadoop.hive.ql.exec.Description;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedExpressions;
+import org.apache.hadoop.hive.ql.exec.vector.expressions.gen.FuncTanhDoubleToDouble;
+import org.apache.hadoop.hive.ql.exec.vector.expressions.gen.FuncTanhLongToDouble;
+import org.apache.hadoop.hive.serde2.io.DoubleWritable;
+
+@Description(name = "tanh",
+        value = "_FUNC_(x) - returns the tangent hyperbolic of x (x is in radians)",
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/udf/UDFTanh.java:28]: returns the `hyperbolic tangent`?

File: vector-code-gen/src/org/apache/hadoop/hive/tools/GenVectorCode.java [hunk 4]
```diff
@@ -1100,10 +1100,16 @@ public class GenVectorCode extends Task {
       {"ColumnUnaryFunc", "FuncAbs", "long", "long", "MathExpr.abs", "", "", "", ""},
       {"ColumnUnaryFunc", "FuncSin", "double", "double", "Math.sin", "", "", "", ""},
       {"ColumnUnaryFunc", "FuncSin", "double", "long", "Math.sin", "(double)", "", "", ""},
+            {"ColumnUnaryFunc", "FuncSinh", "double", "double", "StrictMath.sinh", "", "", "", ""},
```
Comment (deniskuzZ) [inline vector-code-gen/src/org/apache/hadoop/hive/tools/GenVectorCode.java:1103]: please check the line formatting

**Review by deniskuzZ — COMMENTED** [review]
LGTM, thank you for submitting the PR @ggangadharan. I've added some minor comments, please take a look

File: ql/src/java/org/apache/hadoop/hive/ql/udf/UDFSinh.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedExpressions;
+import org.apache.hadoop.hive.ql.exec.vector.expressions.gen.FuncSinhDoubleToDouble;
+import org.apache.hadoop.hive.ql.exec.vector.expressions.gen.FuncSinhLongToDouble;
+import org.apache.hadoop.hive.serde2.io.DoubleWritable;
+
+/**
+ * UDFSinh.
+ */
+@Description(name = "sinh",
+        value = "_FUNC_(x) - returns the Sin hyperbolic  of x (x is in radians)",
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/udf/UDFSinh.java:31]: returns `hyperbolic sine`?

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/Vectorizer.java [hunk 6]
```diff
@@ -51,6 +51,7 @@
 import org.apache.hadoop.hive.ql.exec.vector.reducesink.*;
 import org.apache.hadoop.hive.ql.exec.vector.udf.VectorUDFArgDesc;
 import org.apache.hadoop.hive.ql.io.AcidUtils;
+import org.apache.hadoop.hive.ql.udf.*;
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/Vectorizer.java:54]: wildcard imports

File: ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java (hunk 1 above)
Comment (ggangadharan) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java]: Removed wildcard import and imported only necessary newly added UDF classes

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/Vectorizer.java (hunk 6 above)
Comment (ggangadharan) [reply ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/Vectorizer.java]: Removed wildcard import and imported only necessary newly added UDF classes

File: ql/src/java/org/apache/hadoop/hive/ql/udf/UDFCosh.java (hunk 2 above)
Comment (ggangadharan) [reply ql/src/java/org/apache/hadoop/hive/ql/udf/UDFCosh.java]: Updated the return statement and necessary q.out files in the new commit

File: ql/src/java/org/apache/hadoop/hive/ql/udf/UDFTanh.java (hunk 3 above)
Comment (ggangadharan) [reply ql/src/java/org/apache/hadoop/hive/ql/udf/UDFTanh.java]: Updated the return statement and necessary q.out files in the new commit

File: ql/src/java/org/apache/hadoop/hive/ql/udf/UDFSinh.java (hunk 5 above)
Comment (ggangadharan) [reply ql/src/java/org/apache/hadoop/hive/ql/udf/UDFSinh.java]: Updated the return statement and necessary q.out files in the new commit

File: vector-code-gen/src/org/apache/hadoop/hive/tools/GenVectorCode.java (hunk 4 above)
Comment (ggangadharan) [reply vector-code-gen/src/org/apache/hadoop/hive/tools/GenVectorCode.java]: formatted the lines in the new commit

**Review by deniskuzZ — APPROVED** [review]
LGTM +1, pending tests

**Comment by ggangadharan** [comment]
@deniskuzZ  Thanks for the code review. Looks like the tests are passed now.

**Pull request merged**